src/lib/import/import-utils.ts

```diff
@@ -228,8 +228,16 @@ export class ImportUtils {
     }
 
     public attachReference(arc: Arc<NodeElement, NodeElement>, reference: Place | DataVariable): void {
-        const weight = reference instanceof Place ? reference.marking : parseInt(reference.init?.value ?? '' as string, 10);
-
+        let weight: number;
+        if (reference instanceof Place) {
+            weight = reference.marking;
+        } else {
+            if (!!reference.init && !!reference.init.value && /^[1-9]\d*$/.test(reference.init.value.trim())) {
```

Comment:
This line is also used in Builder, please refactor it into two static functions of ImportUtils. One for the whole line and one for just the regex test because it is also used separately in Builder.

Reply:
Also should work with values like "1.0"